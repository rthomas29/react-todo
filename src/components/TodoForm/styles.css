input {
  margin-left: 5px;
  margin-right: 5px;
}

input:focus {
  outline: none;
}

.add-btn {
  height: 31px;
  border-radius: 5px;
  background-color: #6bb9f0;
  color: white;
  font-size: 1em;
}