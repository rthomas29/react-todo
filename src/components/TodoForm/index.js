import React, { useState } from 'react'
import Button from '../Button'
import './styles.css'

const TodoForm = ({ handleTaskAdd, handleTaskDelete }) => {
  const [inputState, setInputState] = useState('')
  const handleTaskSubmit = (e) => {
    const value = inputState
    e.preventDefault()
    setInputState('')
    handleTaskAdd(value)
  }
  return (
    <form className='form-inline' onSubmit={(e) => handleTaskSubmit(e)}>
      <div className='form-group'>
        <label htmlFor='task-input'>Task:</label>
        <input
          id='task-input'
          type='text'
          className='form-control'
          placeholder='Enter new task here'
          aria-label='checkbox to toggle task completion'
          onChange={(e) => setInputState(e.target.value)}
          value={inputState}
          autoFocus
        />
        <Button type='submit' className='btn btn-success'>Add</Button>
      </div>
    </form>
  )
}

export default TodoForm