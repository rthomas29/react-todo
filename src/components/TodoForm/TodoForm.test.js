import React from 'react'
import { render } from 'react-testing-library'
import TodoForm from './index'

describe('<TodoForm />', () => {
  it('should render', () => {
    const { container } = render(<TodoForm />)
    expect(container.firstChild).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})