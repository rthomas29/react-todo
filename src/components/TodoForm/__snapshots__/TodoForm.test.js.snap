// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm /> should match snapshot 1`] = `
<form
  class="form-inline"
>
  <div
    class="form-group"
  >
    <label
      for="task-input"
    >
      Task:
    </label>
    <input
      aria-label="checkbox to toggle task completion"
      class="form-control"
      id="task-input"
      placeholder="Enter new task here"
      type="text"
      value=""
    />
    <button
      class="btn btn-success"
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
